src/Serilog.Settings.Combined/Debugging/CombinedSetingsBuilderDebuggingExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using Serilog.Settings.Combined;
+
+namespace Serilog.Debugging
+{
+    /// <summary>
+    /// Extensions to help in debugging Combined settings issues
+    /// </summary>
+    public static class CombinedSetingsBuilderDebuggingExtensions
+    {
+        /// <summary>
+        /// Allows to inspect the results of settings combination
+        /// </summary>
+        /// <param name="self">The builder to allow calling in a fluent way</param>
+        /// <param name="action">An action that will be called with a copy of the list of registered settings</param>
+        /// <returns>An object allowing configuration to continue.</returns>
+        public static ICombinedSettingsBuilder Inspect(this ICombinedSettingsBuilder self, Action<IEnumerable<KeyValuePair<string, string>>> action)
+        {
+            if (self == null) throw new ArgumentNullException(nameof(self));
+            if (action == null) throw new ArgumentNullException(nameof(action));
+
+            if (self is CombinedSettingsBuilder concreteBuilder)
+            {
+                var settings = concreteBuilder.BuildSettings();
+                action(new List<KeyValuePair<string, string>>(settings));
+            }
+            else
+            {
+                SelfLog.WriteLine($"{nameof(Inspect)} called on a {nameof(ICombinedSettingsBuilder)} that is not a {nameof(CombinedSettingsBuilder)}. Cannot dump the key-value pairs. Concrete type is {self.GetType()}");
```

Comment:
Another thought, do we need the interface here, or would just using the concrete type throughout eliminate this type of issue?

Reply:
Do you mean getting rid of the `ICombinedSettingsBuilder` interface all together ? That would mean making the concrete calss `CombinedSettingsBuilder` public and `sealed`too, I guess ... I don't have strong opinions on that, just that using interfaces seemed *more pure* when doing fluent-interface-y stuff :)

Reply:
and make the constructor `internal` also.

Reply:
That was the gist of what I had in mind, but could be pros/cons. Classes and/or ABCs do generally evolve a bit better (it's not a breaking change to add a method, for example), but could be other reasons to prefer an interface here - just an idea :-)